pkg/registration/spoke/spokeagent.go

```diff
@@ -359,7 +359,10 @@ func (o *SpokeAgentConfig) RunSpokeAgentWithSpokeInformers(ctx context.Context,
 		if err := wait.PollUntilContextCancel(bootstrapCtx, 1*time.Second, true, o.internalHubConfigValidFunc); err != nil {
 			// TODO need run the bootstrap CSR forever to re-establish the client-cert if it is ever lost.
 			stopBootstrap()
-			return fmt.Errorf("failed to wait for hub client config for managed cluster to be ready: %w", err)
+			// DO NOT return error if context is canceled, allows the leader to release the leadership.
```

Comment:
looks good in general.

Just a question: do we need to doc in which case internalHubConfigValidFunc should return err and it means the agent return with error, and will start a new leader election?

Reply:
yes, more comments added.